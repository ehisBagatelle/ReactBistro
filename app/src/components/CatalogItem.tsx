import React from "react";
import { TouchableOpacity, Text } from "react-native";
import { LineItem } from "../services/OrderService";
import { formatDecimal } from "../helper";
import { styles } from "./Components.styles";

interface Props {
  item: LineItem;
  onPress: () => void;
}

const CatalogItem: React.FC<Props> = ({ item, onPress }) => (
  <TouchableOpacity onPress={onPress} style={styles.item}>
    <Text style={styles.title}>{item.name}</Text>
    <Text style={styles.title}>${formatDecimal(item.price)}</Text>
  </TouchableOpacity>
);

export default CatalogItem;
