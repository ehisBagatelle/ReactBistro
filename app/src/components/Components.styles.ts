import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  item: {
    padding: 10,
    justifyContent: "space-between",
    display: "flex",
    flexDirection: "row",
    width: "100%",
  },
  title: {
    fontSize: 18,
    fontStyle: "normal",
    fontWeight: "400",
  },
  header: {
    fontSize: 32,
    backgroundColor: "#fff",
  },

  backTextWhite: {
    color: "#FFF",
  },
  rowFront: {
    alignItems: "center",
    backgroundColor: "#FFF",

    justifyContent: "center",
    height: 50,
  },
  rowBack: {
    alignItems: "center",
    backgroundColor: "#DDD",
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    paddingLeft: 15,
  },
  backRightBtn: {
    alignItems: "center",
    bottom: 0,
    justifyContent: "center",
    position: "absolute",
    top: 0,
    width: 75,
  },
  backRightBtnLeft: {
    backgroundColor: "blue",
    right: 75,
  },
  backRightBtnRight: {
    backgroundColor: "red",
    right: 0,
  },
  discountAndBill: {
    width: "55%",
    alignSelf: "flex-end",
  },
  discount: {},
  bill: {
    padding: 15,
    width: "100%",
  },
  billItem: {
    display: "flex",
    justifyContent: "space-between",
    flexDirection: "row",
  },
  billItemText: {
    fontSize: 22,
    fontStyle: "normal",
    fontWeight: "400",
  },
  billItemAmount: {
    fontSize: 22,
    fontStyle: "normal",
    fontWeight: "700",
  },
});
