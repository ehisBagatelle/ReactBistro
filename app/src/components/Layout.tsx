import React, { ReactNode } from "react";
import { Text, View, Image, ScrollView } from "react-native";
import { StyleSheet } from "react-native";


interface Props {
  children?: ReactNode;
}

const Layout: React.FC<Props> = ({ children }) => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Image source={require("../../assets/icons8-bistro-64.png")} />
        <Text style={styles.logoText}>React Bistro</Text>
      </View>
      <View style={{ width: "100%" }}>{children}</View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "flex-start",
    width: '100%'
  },
  header:{
    display:'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent:'flex-start',
    width: '100%',
    paddingVertical: 10
  },
  logoText:{
    color: "#000",
    fontSize: 60,
    fontStyle: "normal",
    fontWeight: "400",
  }
});

export default Layout;
