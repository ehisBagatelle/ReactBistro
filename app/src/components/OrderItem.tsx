import React from "react";
import { TouchableHighlight, View, Text } from "react-native";
import { LineItem } from "../services/OrderService";
import { styles } from "./Components.styles";
import { formatDecimal } from "../helper";

interface Props {
  item: LineItem;
}

const OrderItem: React.FC<Props> = ({ item }) => (
  <View style={styles.rowFront}>
    <View style={styles.item}>
      <Text style={styles.title}>{item.name}</Text>
      <Text style={styles.title}>${formatDecimal(item.price)}</Text>
    </View>
  </View>
);

export default OrderItem;
