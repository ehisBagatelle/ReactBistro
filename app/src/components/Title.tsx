import React from "react";
import {
  Text,
  View,
  StyleSheet,
  Image,
  ImageSourcePropType,
} from "react-native";

interface Props {
  name: string;
  iconSource?: ImageSourcePropType;
}

export const Title = ({ name, iconSource }: Props) => {
  return (
    <View style={styles.box}>
      <Text>{name}</Text>
      {iconSource && <Image source={iconSource} />}
    </View>
  );
};

export const styles = StyleSheet.create({
  box: {
    width: "100%",
    height: 45,
    backgroundColor: "#F5F5F5",
    justifyContent: "space-between",
    alignItems: "center",
    display: 'flex',
    flexDirection: 'row',
    paddingHorizontal: 10,
  },
});
