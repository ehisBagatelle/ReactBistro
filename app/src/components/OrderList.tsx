import React, { useState } from "react";
import { SwipeListView } from "react-native-swipe-list-view";
import OrderItem from "./OrderItem";
import { LineItem } from "../services/OrderService";
import { styles } from "./Components.styles";
import { TouchableOpacity, View, Text } from "react-native";

interface Props {
  lineItems: LineItem[];
  addOrRemoveLineItem: (item: LineItem) => void;
}

const OrderList: React.FC<Props> = ({ lineItems, addOrRemoveLineItem }) => {
  const closeRow = (
    rowMap: { [x: string]: { closeRow: () => void } },
    rowKey: string
  ) => {
    if (rowMap[rowKey]) {
      rowMap[rowKey].closeRow();
    }
  };

  const deleteRow = (rowMap: any, lineItem: LineItem) => {
    closeRow(rowMap, lineItem.key);
    addOrRemoveLineItem(lineItem);
  };

  const onRowDidOpen = (rowKey: string) => {
    console.log("This row opened", rowKey);
  };

  const renderHiddenItem = (
    data: { item: LineItem },
    rowMap: { [x: string]: { closeRow: () => void } }
  ) => {
    return (
      <View style={styles.rowBack}>
        <TouchableOpacity
          style={[styles.backRightBtn, styles.backRightBtnLeft]}
          onPress={() => closeRow(rowMap, data.item.key)}
        >
          <Text style={styles.backTextWhite}>Close</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles.backRightBtn, styles.backRightBtnRight]}
          onPress={() => deleteRow(rowMap, data.item)}
        >
          <Text style={styles.backTextWhite}>Delete</Text>
        </TouchableOpacity>
      </View>
    );
  };

  return (
    <SwipeListView
      data={lineItems}
      renderItem={(data) => <OrderItem item={data.item} />}
      renderHiddenItem={renderHiddenItem}
      leftOpenValue={75}
      rightOpenValue={-150}
      previewRowKey={"0"}
      previewOpenValue={-40}
      previewOpenDelay={3000}
      onRowDidOpen={onRowDidOpen}
    />
  );
};

export default OrderList;
