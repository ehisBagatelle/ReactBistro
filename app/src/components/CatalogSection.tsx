import React from "react";
import { SectionList } from "react-native";
import CatalogItem from "./CatalogItem";
import { Title } from "../components/Title";
import { LineItem } from "../services/OrderService";

interface Props {
  sections: {
    title: string;
    data: LineItem[];
  }[];
  onCatalogItemPress: (item: LineItem) => void;
}

const CatalogSection: React.FC<Props> = ({ sections, onCatalogItemPress }) => {
  return (
    <SectionList
      sections={sections}
      keyExtractor={(item, index) => item.name + index}
      renderItem={({ item }) => (
        <CatalogItem item={item} onPress={() => onCatalogItemPress(item)} />
      )}
      renderSectionHeader={({ section: { title } }) => <Title name={title} />}
    />
  );
};

export default CatalogSection;
