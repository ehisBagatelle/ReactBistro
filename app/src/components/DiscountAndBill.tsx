import React from "react";
import { View, TouchableOpacity, Text } from "react-native";
import { styles } from "./Components.styles";
import { Title } from "./Title";
import { LineItem } from "../services/OrderService";

interface Props {
  onPressDiscount: () => void;
  order: LineItem[];
}

const DiscountAndBill: React.FC<Props> = ({ onPressDiscount, order }) => (
  <View style={styles.discountAndBill}>
    <View style={styles.discount}>
      <TouchableOpacity onPress={onPressDiscount}>
        <Title
          name="Tap To Add Discount"
          iconSource={require("../../assets/icons8-discount-50.png")}
        />
      </TouchableOpacity>
    </View>
    <View style={styles.bill}>
      <View style={styles.billItem}>
        <Text style={styles.billItemText}>Subtotal</Text>
        <Text style={styles.billItemAmount}>$12.99</Text>
      </View>
      <View style={styles.billItem}>
        <Text style={styles.billItemText}>Discounts</Text>
        <Text style={styles.billItemAmount}>$12.99</Text>
      </View>
      <View style={styles.billItem}>
        <Text style={styles.billItemText}>Tax</Text>
        <Text style={styles.billItemAmount}>$12.99</Text>
      </View>
      <View style={styles.billItem}>
        <Text
          style={styles.billItemText}
        >{`Total (${order.length} item)`}</Text>
        <Text style={styles.billItemAmount}>$12.99</Text>
      </View>
    </View>
  </View>
);

export default DiscountAndBill;
