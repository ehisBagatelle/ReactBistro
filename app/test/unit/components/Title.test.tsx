import React from "react";
import renderer from "react-test-renderer";
import { Title } from "../../../src/components/Title";

describe("Title", () => {
  it("render title with only name", () => {
    const tree = renderer.create(<Title name="title" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("render title with name and icon", () => {
    const tree = renderer
      .create(
        <Title name="title" iconSource={require("../../../assets/icon.png")} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
