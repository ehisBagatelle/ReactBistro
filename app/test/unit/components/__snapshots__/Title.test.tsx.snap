// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title render title with name and icon 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F5F5F5",
      "display": "flex",
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <Text>
    title
  </Text>
  <Image
    source={
      {
        "testUri": "../../../assets/icon.png",
      }
    }
  />
</View>
`;

exports[`Title render title with only name 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F5F5F5",
      "display": "flex",
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <Text>
    title
  </Text>
</View>
`;
